internal/statemachine/helper.go

```diff
@@ -657,3 +657,40 @@ func generateDebootstrapCmd(imageDefinition ImageDefinition, targetDir string, i
 
 	return debootstrapCmd
 }
+
+// generateAptCmd generates the apt command used to create a chroot
+// environment that will eventually become the rootfs of the resulting image
+func generateAptCmd(targetDir string, packageList []string) *exec.Cmd {
+	aptCmd := execCommand("chroot", targetDir, "apt", "install", "-y")
+
+	for _, aptPackage := range packageList {
+		aptCmd.Args = append(aptCmd.Args, aptPackage)
+	}
+
+	aptCmd.Env = append(aptCmd.Env, "DEBIAN_FRONTEND=noninteractive")
+
+	return aptCmd
+}
+
+// createPPAInfo generates the name for a PPA sources.list file
+// in the convention of add-apt-repository, and the contents
+// that define the sources.list
+// TODO: figure out what to do with fingerprint
+func createPPAInfo(ppa *PPAType, series string) (fileName string, fileContents string) {
```

Comment:
I wonder if this should create a deb822 style source. They're the future!

They can also contain signing key directly, which avoids the issue of having to figure out where on disk to put that.